File: JavaScript/429-N-ary Tree Level Order Traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if (!root) return [];
    
    let res = [], queue = [root];
    
    while (queue.length){
        let level = [], queueL = queue.length;
        for (let i=0; i< queueL; i++){
            let curr = queue.shift();
            level.push(curr.val);
            for (let nxt of curr.children){
                queue.push(nxt);
            }
        }
        res.push(level);
    }
    return res;
};

File: 429-N-ary Tree Level Order Traversal.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""
from collections import deque
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: Node
        :rtype: List[List[int]]
        """
        if not root:
            return []
        
        res = []
        
        queue = deque([root])
        
        while queue:
            level = []
            for _ in range(len(queue)):
                curr = queue.popleft()
                level.append(curr.val)
                for nxt in curr.children:
                    queue.append(nxt)
            res.append(level)
            
        return res